command_test.go

```diff
@@ -486,3 +486,15 @@ func TestPersistentRequiredFlags(t *testing.T) {
 		}
 	}
 }
+
+// TestUpdateName checks if c.Name() updates on changed c.Use.
+// Related to https://github.com/spf13/cobra/pull/422#discussion_r143918343.
+func TestUpdateName(t *testing.T) {
+	c := &Command{Use: "name xyz"}
+	originalName := c.Name()
+
+	c.Use = "changedName abc"
+	if originalName == c.Name() {
```

Comment:
Is this the right test? Should it be:
```
if c.Name() != "changedName" {
```

Reply:
@eparis The main purpose here is to see whether `c.Name()` gets changed after assigning a new value to `c.Use`.

We may add a extra statement here to see if `c.Name() != "changedName"`.